fluent_pose_synthesis/data/load_data.py

```diff
@@ -0,0 +1,203 @@
+            disfluent_file = fluent_file.with_name(fluent_file.name.replace("_original.pose", "_updated.pose"))
+            metadata_file = fluent_file.with_name(fluent_file.name.replace("_original.pose", "_metadata.json"))
+            # Load pose objects
+            with open(fluent_file, 'rb') as f:
+                fluent_pose = Pose.read(f.read())
+            with open(disfluent_file, 'rb') as f:
+                disfluent_pose = Pose.read(f.read())
+            with open(metadata_file, 'r', encoding='utf-8') as f:
+                metadata = json.load(f)
+            self.examples.append({
+                'fluent': fluent_pose,
+                'disfluent': disfluent_pose,
+                'metadata': metadata
+            })
+            # Record the frame count for this sample
+            min_length = min(fluent_pose.body.data.shape[0], disfluent_pose.body.data.shape[0])
+            frame_nums_list.append(min_length)
+
+        # Generate sliding window indices using vectorized method
+        # Create a list to hold indices arrays for each sample
+        self.clip_indices_list = []
+        sample_idx = 0
+        for i, example in enumerate(self.examples):
+            # Get the number of frames in the fluent sequence; Skip if the sequence is too short
+            min_length = min(example['fluent'].body.data.shape[0], example['disfluent'].body.data.shape[0])
+            if min_length < self.window_size:   
+                continue
+            # Generate sliding window start indices with the given offset
+            clip_indices = np.arange(0, min_length - self.window_size + 1, self.offset_frame)[:, None] + np.arange(self.window_size)
+            # Prepend the sample index (sample_idx) to each clip index row
+            clip_indices_with_idx = np.hstack((np.full((len(clip_indices), 1), i, dtype=clip_indices.dtype), clip_indices))
+            self.clip_indices_list.append(clip_indices_with_idx)
+            sample_idx += 1
+        # Concatenate all clip indices from different samples
+        self.clip_indices = np.concatenate(self.clip_indices_list, axis=0)
+
+        total_clips = len(self.clip_indices)
+        total_frames = sum(frame_nums_list)
+        print('Dataset loaded, trained with %d clips, %d frames, %d mins in total' % (total_clips, total_frames, total_frames/25/60))   # All sequences are sampled at 25fps
+
+    def __len__(self) -> int:
+        """
+        Returns the total number of sliding-window clips in the dataset.
+        """
+        return len(self.clip_indices)
+
+    def __getitem__(self, idx: int) -> Dict[str, Any]:
+        """
+        Retrieves a sliding-window clip from the dataset.
+        Args:
+            idx (int): Index of the clip to retrieve.
+        """
+        # Each row of clip_indices: [example_index, frame0, frame1, ..., frame(window_size-1)]
+        indices = self.clip_indices[idx]
```

Comment:
awesome. so what you are doing right now is creating a "pre-defined" list of clips and indices.
How about instead, we can say that "getitem" gets a clip index (so you just have a list of all clips)
and then chooses a "random" window to train on?

that would mean that in every epoch, each sample will be seen once, in one window of it.